File: src/main/java/org/example/blind/TwoSum.java
package org.example.blind;

import java.util.HashMap;

public class TwoSum {

    public int[] twoSum(int[] nums, int target) {
        int[] result = new int[2];
        HashMap<Integer, Integer> map = new HashMap<>();

        for (int i = 0; i < nums.length; i++) {
            int diff = target - nums[i];
            if (map.containsKey(diff)) {
                result[0] = map.get(diff);
                result[1] = i;
            }
            map.put(nums[i], i);
        }

        return result;
    }
    public int[] twoSumO_N_Square(int[] nums, int target) {
        int[] result = new int[2];
        for (int i = 0; i < nums.length; i++) {
            for(int j = i+1; j < nums.length; j++) {
                if (nums[i] + nums[j] == target) {
                    result[0] = i;
                    result[1] = j;
                    break;
                }
            }
        }

        return result;
    }
}

File: src/main/java/org/example/blind/TwoSum.py
class Solution:
    def twoSum(self, nums, target):
        hmap = {}

        for i, n in enumerate(nums):
            diff = target - n
            if diff in hmap:
                return [hmap[diff], i]
            hmap[n] = i
        return []

obj = Solution()
#nums = [3,2,4]
#nums = [2,7,11,15]
nums = [3,3]
print(obj.twoSum(nums, 6))
